src/arm-codegen.c

```diff
@@ -421,6 +427,22 @@ void emit_ph2_ir(ph2_ir_t *ph2_ir)
         emit(__mov_i(__NE, rd, 0));
         emit(__mov_i(__EQ, rd, 1));
         return;
+    case OP_trunc:
+        if (rm == 1) {
+            rm = 0xFF;
+        } else if (rm == 4) {
+            rm = 0xFFFFFFFF;
+        } else {
+            printf("Unsupported truncation operation with target size %d\n",
```

Comment:
Use `error` when possible.

Reply:
The issue of using error here is that it would leads to incorrect error diagnostic location, which is kind of meaningless?

Reply:
I would like to open a new issue regards this issue since `error` is not capable of passing variadic arguments and it would make error message being limited.

Reply:
> I would like to open a new issue regards this issue since `error` is not capable of passing variadic arguments and it would make error message being limited.

Go ahead.